### DISCUSSION TITLE
Rsubmit and SASpy

### DISCUSSION THREAD

**Discussion by ifly6** [sassoftware/saspy#386] [Q&A]
I don't know where to ask this (I'd normally go with StackOverflow but it seems that isn't in the cards here).

If I have a SAS script which establishes certain commands to be `rsubmit`ted to a remote server, is there an analogous way to do this in SASpy, or does the library only support, currently, local submissions?

Thanks!

**Comment by tomweber-sas** [comment]
saspy supports connecting to SAS locally as well as remotely. To various kinds of deployments, from a simple install, to a Production deployment with workspace servers and metadata, to Viya, either V3 or V4 in the cloud. It has different access methods for the various kinds of deployments. There's a 'chart', sorta, that shows you which one you can use depending upon SAS and where your client is, here in the doc: https://sassoftware.github.io/saspy/install.html#choosing-an-access-method
That's in the config section of the doc, so once you know how you would be connecting to the SAS you want to conect to (fromwhere), you can go to that section to see how to configure it. Usually only takes a few config options, but as it's SAS, and there can be a lot of extra features, there are many extra config options. But, don't you usually just need the couple of required one, like 'where is SAS?' host/port, things like that.
Take a look and see if that makes sense. 
Thanks,
Tom

I guess the other thing you might be asking is if you are connected locally, can you submit those rsubmit (SAS/Connect?) from that session to then go run somewhere else? Yes, you can do that too. Again, obviously, you have to be able to connect from the local SAS you're connect to to that remote one, but that's already the case. That's just SAS code at that point; saspy wouldn't know about it.

Is that the question?
Tom
